cluster/schema/meta_class.go

```diff
@@ -478,6 +480,20 @@ func (m *metaClass) applyShardProcess(name string, action command.TenantProcessR
 
 		if count == len(processes) {
 			copy.Status = req.Tenant.Status
+			// Whenever a tenant is frozen, we send a notification the querier nodes registered
+			// with the querier manager. This call to NotifyClassTenantDataUpdate is non-blocking
```

Comment:
`NotifyClassTenantDataUpdate` is actually blocking because it expects to acquire a lock on the querier map under the hood. See my comment below how we could make this non blocking.

Reply:
Good point, I was assuming the lock would never experience contention, thanks for pointing that out